src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -202,16 +206,35 @@ DeparseCreateRoleStmt(Node *node)
 
 
 /*
- * AppendCreateRoleStmt generates the string representation of the
- * CreateRoleStmt and appends it to the buffer.
+ * AppendRoleOption generates the string representation of the DefElem option
+ * and appends it to the buffer.
  */
 static void
-AppendCreateRoleStmt(StringInfo buf, CreateRoleStmt *stmt)
+AppendInlinePriviliges(StringInfo buf, ListCell *optionCell)
 {
-	ListCell *optionCell = NULL;
+	DefElem *option = (DefElem *) lfirst(optionCell);
+
+	if (strcmp(option->defname, "adminmembers") == 0)
+	{
+		appendStringInfo(buf, " ADMIN ");
+		AppendRoleList(buf, (List *) option->arg);
+	}
+	else if (strcmp(option->defname, "rolemembers") == 0)
+	{
+		appendStringInfo(buf, " ROLE ");
+		AppendRoleList(buf, (List *) option->arg);
+	}
+	else if (strcmp(option->defname, "addroleto") == 0)
+	{
+		appendStringInfo(buf, " IN ROLE ");
+		AppendRoleList(buf, (List *) option->arg);
+	}
```

Comment:
I think it would be better if this was part of the `AppendRoleOption` function. The split seems rather arbitrary.

Reply:
So actually they should not be made part of AppendRoleOption. And SYSID should also not be part of it. Since AppendRoleOption is used both for ALTER ROLE and CREATE ROLE. See the comment on AppendRoleOption:
```
 * This function only generates strings for common role options of ALTER ROLE
 * and CREATE ROLE statements. The extra options for CREATE ROLE are handled
 * seperately.
 ```

Reply:
Changed the structure. Could you check it again?